### PR TITLE
update api to paddle 2.1

### PR DESCRIPTION
**Pull request by GuoxiaWang**
本次 PR 主要是将 PLSC 仓库从 Paddle 1.x 升级适配到 Paddle 2.1 API 接口

除了API改变以外，还做了如下几点改进：

* 重新实现了 dist-softmax 和 dist-arcface 的 fp16 混合训练逻辑
* 重新实现了checkpoint 保存和加载的 save，load 函数
* 重新实现了 resume 逻辑，训练停止后，可以直接设置 checkpoint dir 进行自动恢复，不用做任何修改接着训练
* 重新实现了 dist-fc 参数在保存前训练和当前训练GPU数量不一致时，dist-fc 参数重新划分逻辑
* 重新实现了训练阶段的数据加载方式，基于 Dataset 和 DataLoader，对齐动态图使用习惯

### CODE REVIEW

**Review by sandyhouse — APPROVED** [review]
LGTM
